### DISCUSSION TITLE
Streaming Responses

### DISCUSSION THREAD

**Discussion by iamnafets** [trpc/trpc#4469] [Ideas]
I have a use-case where I need streaming responses, e.g. the client makes a request and gets part 1, part 2, part 3 with varying degrees of latency. The traditional way to do this is to have an HTTP request return in chunks which are processed greedily rather than waiting for the whole response. The way I'm doing it in TRPC is opening a websocket subscription, assigning some state to the connection, and then connecting to that state on incoming mutations.

We're running into a TON of issues with long-running websocket connections, load balancers, client-side firewall rules, etc. I'm curious whether TRPC has any concept of these multi-responses or whether you all would be open to them.

**Comment by KATT** [comment]
Feel free to open a feat request with more context, your company can put a bounty on them too :)

**Comment by iamnafets** [comment]
Sorry, stupid question: where do I open a feat request? I'm happy to do some of the work and add a bounty.

**Comment by iamnafets** [comment]
Also: could the same architecture be used to return batches early rather than waiting for the entire batch to complete? Presumably you could stream the response objects.

**Reply by KATT** [reply]
#4347

**Reply by iamnafets** [reply]
You all are ahead of the game!
